### ISSUE TITLE
"RNCSafeAreaProvider" was not found in the UIManager.]

### ISSUE THREAD

**Issue by crukundo** [storybookjs/react-native#422]
**Describe the bug**
```
This error is located at:
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider
    in Unknown (created by StorybookUIRoot)
    in StorybookUIRoot (created by withDevTools(StorybookUIRoot))
    in withDevTools(StorybookUIRoot)
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in main(RootComponent)
 ERROR  Unhandled promise rejection [Invariant Violation: requireNativeComponent: "RNCSafeAreaProvider" was not found in the UIManager.]
```

**To Reproduce**
Please attempt to run the example app [here:](https://github.com/crukundo/etta-ui)

**Expected behavior**
This looks like a new error. I have tried to install react-native-safe-area-context with yarn and via expo and it won't budge. However, I am running an expo managed app so `pod install` wasn't necessary before. But nonetheless when I run `expo run:ios`, the pods are installed. No dice with android either. The error stays.  

**Screenshots**
This is exactly the error here https://github.com/storybookjs/react-native/issues/383#issuecomment-1269995234

**Additional context **
I am currently working with [v6.0.1-beta.12](https://github.com/storybookjs/react-native/releases/tag/v6.0.1-beta.12) and I notice there's a new dependency for react-native-safe-area-context in v6.

**Comment by crukundo** [comment]
**What I have done so far:**
- Install react-native-safe-area-context with expo: `npx expo install react-native-safe-area-context`
- run `expo run:ios` then cancel
- Go to ios folder, run `npx pod-install`
- run `expo run:android` then cancel
- Remove react-native-safe-area-context: `yarn remove react-native-safe-area-context`
- run `yarn start -c`. 

Had to uninstall react-native-area-context because there's a new error: _Invariant Violation: Tried to register two views with the same name RNCSafeAreaProvider_. The second last step was important. I am still not sure why this workaround works and it's possible it's just me but I'm unblocked

**Comment by dannyhw** [comment]
@crukundo this usually happens when you have multiple versions of react-native-safe-area-context installed so it might be some sort of metro module resolution issue, its most common in a mono-repo where metro is resolving the package from another folder in the repo.

I've gotten past this in my projects by using rnx-kit to de-dupe modules https://microsoft.github.io/rnx-kit/docs/tools/metro-config#ensuring-a-single-instance-of-a-package

**Comment by flexbox** [comment]
hey @crukundo  If your problem is solved can you please close the issue? 
It sparks joy for maintainers ✨

**Comment by Yashveer-Soni** [comment]
issue occur after i update react-native-safe-area-context

here is my react native info

info Fetching system and libraries information...
System:
  OS: macOS 15.7.3
  CPU: (10) arm64 Apple M4
  Memory: 113.42 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.19.6
    path: /usr/local/bin/node
  Yarn: Not Found
  npm:
    version: 11.7.0
    path: /Users/sunilsharma/Documents/GitHub/ManagementAppNeww/node_modules/.bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.16.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 25.2
      - iOS 26.2
      - macOS 26.2
      - tvOS 26.2
      - visionOS 26.2
      - watchOS 26.2
  Android SDK:
    API Levels:
      - "34"
      - "35"
      - "36"
    Build Tools:
      - 28.0.3
      - 29.0.0
      - 29.0.1
      - 29.0.2
      - 29.0.3
      - 30.0.0
      - 30.0.1
      - 35.0.0
      - 36.1.0
    System Images:
      - android-36.1 | Google Play ARM 64 v8a
      - android-36.1 | Google Play Intel x86_64 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: 2025.2 AI-252.27397.103.2522.14514259
  Xcode:
    version: 26.2/17C52
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.17
    path: /opt/homebrew/opt/openjdk@17/bin/javac
  Ruby:
    version: 3.4.8
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: 15.0.1
  react:
    installed: 19.1.0
    wanted: 19.1.0
  react-native:
    installed: 0.80.1
    wanted: 0.80.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false

info React Native v0.83.0 is now available (your project is running on v0.80.1).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.83.0
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.80.1&to=0.83.0
info For more info, check out "https://reactnative.dev/docs/upgrading?os=macos".

**Comment by dannyhw** [comment]
@Yashveer-Soni A few questions:
What version of safearea? 
Do you have a monorepo? 
Did you try any other solutions? 
Can you provide a reproduction of the problem?
